code/game/dna/dna_modifier.dm

```diff
@@ -787,6 +798,27 @@
 			connected.beaker = null
 		return 1
 
+	if(href_list["changeBlockLabel"])
+		var/which = text2num(href_list["changeBlockLabel"])
+		var/datum/block_label/label = labels[which]
+		var/text = copytext(sanitize(input(usr, "New Label:", "Edit Label", label.name) as text|null),1,MAX_NAME_LEN)
```

Comment:
Should check appropriateness for this after the inputs, due to Adjacency/deadness/power off.

It would be better if you did both inputs without any code in-between so you can check after both and then assign.